### PR TITLE
Remove more dead native defines

### PR DESCRIPTION
**Pull request by danmosemsft**
Contributes to https://github.com/dotnet/coreclr/issues/9067

Still a few more to go.

### CODE REVIEW

**Comment by stephentoub** [comment]
@dotnet-bot test this please (Jenkins was down)

File: src/mscorlib/src/System.Private.CoreLib.txt [hunk 1]
```diff
@@ -1198,40 +1184,6 @@ InvalidOperation_APIInvalidForCurrentContext = The API '{0}' cannot be used on t
 InvalidProgram_Default = Common Language Runtime detected an invalid program.
 
 ; Isolated Storage
```
Comment (justinvp) [inline src/mscorlib/src/System.Private.CoreLib.txt]: This line should be deleted

**Comment by danmosemsft** [comment]
@jkotas PTAL .. although I should have time to finish the last ones tonight, so feel free to wait.

File: src/mscorlib/src/System.Private.CoreLib.txt (hunk 1 above)
Comment (danmosemsft) [inline src/mscorlib/src/System.Private.CoreLib.txt]: done

**Comment by danmosemsft** [comment]
OK this fixes #9067.

**Comment by danmosemsft** [comment]
@dotnet-bot test centos7.1 x64 debug build and test (https://github.com/dotnet/coreclr/issues/9623)

**Comment by danmosemsft** [comment]
@jkotas ready for review. After this I move back to corelib.

**Pull request merged**